File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix=strs[0];
        int n=strs.size();
        for(int i=0;i<n;i++){
            while(!strs[i].starts_with(prefix)){
                prefix=prefix.substr(0,prefix.size()-1);
            }
        }
        return prefix;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        StringBuilder res=new StringBuilder();
        int n=strs.length;
        String first=strs[0];
        String second=strs[n-1];
        for(int i=0;i<first.length();i++){
            if(first.charAt(i)==second.charAt(i)){
                res.append(first.charAt(i));
            }else{
                break;
            }
        }
        return res.toString();

    }
}